### PR TITLE
[DTRA] Ahmad/DTRA-902/Fix blank screen on toggling chart type and interval

### PR DESCRIPTION
**Pull request by ahmadtaimoor-deriv**
This PR addresses a critical issue where users encountered a blank screen upon attempting to toggle between different chart types and intervals within our application. This bug significantly impacted user experience by obstructing data visualization and analysis capabilities.

### CODE REVIEW

File: src/components/Timeperiod.tsx [hunk 1]
```diff
@@ -10,60 +10,57 @@ import { InlineLoader } from './Loader';
 import Tooltip from './Tooltip';
 
 type TTimeperiodItemProps = {
-    category: typeof Intervals[0];
+    category: (typeof Intervals)[0];
```
Comment (kate-deriv) [inline src/components/Timeperiod.tsx:13]: Are brackets needed here? Please, ignore if not relevant

File: src/components/Timeperiod.tsx [hunk 2]
```diff
@@ -10,60 +10,57 @@ import { InlineLoader } from './Loader';
...
-    const { timeperiod, chartType, loader, chart, state } = useStores();
+const TimeperiodItemComponent = ({ item, category, onClick, isLoading }: TTimeperiodItemProps) => {
+    const { timeperiod, chartType, chart, state } = useStores();
     const chartTypeId = chartType.type.id;
     const { allowTickChartTypeOnly } = state;
     const { mainStore } = timeperiod;
     const { granularity } = chart;
     const isMobile = mainStore.chart.isMobile as boolean;
-    const { isActive: isLoading } = loader;
+    // const { isActive: isLoading } = loader;
```
Comment (farhan-nurzi-deriv) [inline src/components/Timeperiod.tsx:34]: Maybe you can remove this

Comment (kate-deriv) [inline src/components/Timeperiod.tsx:34]: Can we delete it if not needed?

File: src/components/Timeperiod.tsx (hunk 1 above)
Comment (maryia-deriv) [reply src/components/Timeperiod.tsx:13]: without brackets TS will try to take the first element if `Intervals`. but it needs to take the first element of `typeof Intervals`. Is that right  @ahmadtaimoor-deriv  ? :)

Comment (ahmadtaimoor-deriv) [reply src/components/Timeperiod.tsx]: The one Kate mentioned typeof Intervals[0] works the same as with bracket. I removed it , thanks

**Review by kate-deriv — APPROVED** [review]

**Review by maryia-deriv — APPROVED** [review]

**Review by henry-deriv — APPROVED** [review]

**Pull request merged**